tribe-ext-ea-additional-options.php

```diff
@@ -0,0 +1,329 @@
+                }
+                $fieldMap['unsafe_description'] = 'post_content';
+            }
+            return $fieldMap;
+        }
+
+        /**
+         * Filters event info before being saved or updated
+         * 
+         * @param array $event
+         * @return array
+         */
+        public function filter_imported_event($event, $record) {
+            $meta = $record->meta;
+            $lineBreakOpt = tribe_get_option($this->opts_prefix . 'retain_line_breaks');
+            if ($lineBreakOpt == 'yes') {
+                $event['post_content'] = str_replace(['\\n', '\n', "\n", "\\n"], '<br>', $event['post_content']);
+            }
+            if(!empty($meta['prefix'])){
+                $event['post_title'] = $meta['prefix'] . ' ' . $event['post_title'];
+            }
+            if(!empty($meta['timezone'])){
+                if(!empty($event['EventAllDay']) && $event['EventAllDay'] == 'yes'){
+                    $event['EventTimezone'] = $meta['timezone'];
+                }else{
+                    date_default_timezone_set( "UTC" );
+                    $targetOffset = timezone_offset_get(timezone_open($meta['timezone']), new DateTime());
+                    $eventOffset = timezone_offset_get(timezone_open($event['EventTimezone']), new DateTime());
+                    $currTimezone = new DateTimeZone($event['EventTimezone']);
+                    $currStartDateTime = new DateTime($event['EventStartDate'] . ' ' . $event['EventStartHour'] . ':' . $event['EventStartMinute'], $currTimezone);
+                    $currEndDateTime = new DateTime($event['EventEndDate'] . ' ' . $event['EventEndHour'] . ':' . $event['EventEndMinute'], $currTimezone);
+                    $offsetDiff = intval($targetOffset) - intval($eventOffset);
+                    $targetInterval = DateInterval::createFromDateString((string) $offsetDiff . ' seconds');
+                    $currStartDateTime->add($targetInterval);
+                    $currEndDateTime->add($targetInterval);
+                    $event['EventStartDate'] = $currStartDateTime->format('Y-m-d');
+                    $event['EventStartHour'] = $currStartDateTime->format('H');
+                    $event['EventStartMinute'] = $currStartDateTime->format('i');
+                    $event['EventEndDate'] = $currEndDateTime->format('Y-m-d');
+                    $event['EventEndHour'] = $currEndDateTime->format('H');
+                    $event['EventEndMinute'] = $currEndDateTime->format('i');
+                    $event['EventTimezone'] = $meta['timezone'];
+                }
+            }
+            return $event;
+        }
+
+        public function filter_import_meta($meta) {
+            $post_data = $_POST['aggregator'];
+            $meta['prefix'] = empty($post_data['prefix']) ? '' : sanitize_text_field($post_data['prefix']);
+            $meta['timezone'] = empty($post_data['timezone']) ? null : $post_data['timezone'];
```

Comment:
Same as above adding a sanitization is recommendend. 

```suggestion
            $meta['timezone'] = empty($post_data['timezone']) ? null : sanitize_text_field( $post_data['timezone'] );
```